src/test/mir-opt/const_goto.issue_77355_opt.ConstGoto.diff

```diff
@@ -7,33 +7,35 @@
 -     let mut _2: bool;                    // in scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
 -     let mut _3: isize;                   // in scope 0 at $DIR/const_goto.rs:12:22: 12:28
 +     let mut _2: isize;                   // in scope 0 at $DIR/const_goto.rs:12:22: 12:28
+      scope 1 {
+      }
```

Comment:
I'm not super happy about this, but only creating a `scope` when we actually created bindings would require much more invasive changes since we need to bubble up the scope from `declare_bindings` in `lower_let_expr`

Reply:
The other option would be to create a scope only for `if let` statements _regardless_ of if they have bindings.

Reply:
@nagisa do you consider this to be an issue? I've reduced it to only introduce scopes for `if let` but it can still happen

Reply:
I haven't had an opportunity to give this PR a thorough look yet. I'll need to think about the implications of this approach too, so I can't promise I'll be able to get back to you this week.

Reply:
That's fine, this bug is a blocker for some of my work but I'll be at a conference next week anyways

Reply:
I think it is perfectly fine to create scopes liberally if it serves some purpose (such as simplifying the code) – it seems like it would likely be much more straightfoward to clean this up as a MIR transformation pass after the fact if this ever came up as a sticking point.